Scrabble.java

```diff
@@ -85,9 +107,23 @@ public static void playHand(String hand) {
 			// non-whitespace characters. Whitespace is either space characters, or  
 			// end-of-line characters.
 			String input = in.readString();
-			//// Replace the following break statement with code
-			//// that completes the hand playing loop
-			break;
+			if (!input.equals(".")) {
+				if ((isWordInDictionary(input)) && (MyString.subsetOf(input, hand))) {
+					score += wordScore(input);
+					hand = MyString.remove(hand, input);
+					System.out.println(input + " earned " + wordScore(input) + " points. Score: " + score + " points");
+					System.out.println("");
+				}
+				else if (!MyString.subsetOf(input, hand)) {
+					System.out.println("Invalid word. Try again.");
+					////System.out.println("");
+				}		
+				else if (!isWordInDictionary(input)) {
+					System.out.println("No such word in the dictionary. Try again.");
+					////System.out.println("");
+				}
+			}
+			if (input.equals(".")) break;

```

Comment:
Could be structured better,
line 126 should be first
.. 

then put 2 else if as ifs with continue. 
then run rest ..